neps/nep-0393.md

```diff
@@ -0,0 +1,480 @@
+  data: Mint | Recover | Renew | Revoke | Burn | Ban[] | SoulTransfer;
+}
+
+/// An event minted by the Registry when new SBT is created.
+type Mint {
+  ctr: AccountId;    // SBT Contract recovering the tokens
+  tokens: [[AccountId, u64[]]];  // list of pairs (token owner, TokenId[])
+}
+
+/// An event emitted when a recovery process succeeded to reassign SBT, usually due to account
+/// access loss. This action is usually requested by the owner, but executed by an issuer,
+/// and doesn't trigger Soul Transfer. Registry recovers all tokens assigned to `old_owner`,
+/// hence we don't need to enumerate them.
+/// Must be emitted by an SBT registry.
+type Recover {
+  ctr: AccountId         // SBT Contract recovering the tokens
+  old_owner: AccountId;  // current holder of the SBT
+  new_owner: AccountId;  // destination account.
+}
+
+/// An event emitted when a existing tokens are renewed.
+/// Must be emitted by an SBT registry.
+type Renew {
+  ctr: AccountId;  // SBT Contract renewing the tokens
+  tokens: u64[];  // list of token ids.
+}
+
+/// An event emitted when an existing tokens are revoked.
+/// Revoked tokens should not be listed in a wallet.
+/// Must be emitted by an SBT registry.
+type Revoke {
+  ctr: AccountId;  // SBT Contract revoking the tokens
+  tokens: u64[];  // list of token ids.
+}
+
+/// An event emitted when an existing tokens are burned.
+/// Must be emitted by an SBT registry.
+type Burn {
+  ctr: AccountId;  // SBT Contract burning the tokens
+  tokens: u64[];  // list of token ids.
+}
+
+/// An event emitted when the `account` is banned within the emitting registry.
+/// Registry must add the `account` to a list of accounts that are not allowed to get any SBT
+/// in the future.
+/// Must be emitted by an SBT registry.
+type Ban {
+  account: AccountId;
+}
+
+/// An event emitted when soul transfer is happening: all SBTs owned by `from` are transferred
+/// to `to`, and the `from` account is banned (can't receive any new SBT).
+/// Must be emitted by an SBT registry.
+/// Registry MUST also emit `Ban` whenever the soul transfer happens.
+type SoulTransfer {
+  from: AccountId;
+  to: AccountId;
+}
+```
+
+Whenever a recovery is made in a way that an existing SBT is burned, the `Burn` event MUST be emitted. If `Revoke` burns token then `Burn` event MUST be emitted instead of `Revoke`.
+
+### Example SBT Contract functions
+
+Although the transaction functions below are not part of the SBT smart contract standard (depending on a use case, they may have different parameters), we present here an example interface for SBT issuance and we also provide a reference implementation.
+These functions must relay calls to an SBT registry, which will emit appropriate events.
+We recommend that the all functions related to an event will take an optional `memo: Option<String>` argument for accounting purposes.
+
+```rust
+trait SBT {
+    /// the function should overwrite `metadata.class = class`.
```

Comment:
Why not just remove the `class` argument instead of having to explain that it overrides `metadata.class`?